echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ArtifactPostProcessor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.pipelinetriggers.postprocessors;
+
+import com.netflix.spinnaker.echo.model.Pipeline;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * Post-processor extracts artifacts from a pipline using a supplied Jinja template and adds
+ * these artifacts to the pipeline as received artifacts.
+ * This post-processor is not implemented yet and is currently a no-op.
+ */
+@Component
+@Order(value = 2)
```

Comment:
This is a documented feature of Spring, so should remain stable.  I could think of a couple of ways of not depending on this:
* Define a `getOrdering` on the `PipelinePostProcessor` interface and have the `PipelinePostProcessorHandler` sort based on this
* Have each `PipelinePostProcessor` declare its dependencies and do some kind of topological sort in the constructor for `PipelinePostProcessor` (which would fail in the event of cyclic dependencies)
Does one of those approaches sound reasonable?

Reply:
Discussed offline; implemented an ordering function in the interface.